code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -203,67 +355,61 @@
 			del(M)
 			new/obj/effect/decal/cleanable/blood/oil(get_turf(src))
 
-			stored_comms["metal"] += 15
-			stored_comms["glass"] += 15
-			stored_comms["wood"] += 5
-			stored_comms["plastic"] += 5
+			store_buffer["metal"] += 15
+			store_buffer["glass"] += 15
+			store_buffer["plastic"] += 15
+			grabbed_something = 1
+			continue
 
-			return
 		else
 			continue
 
+
+
 	for(var/obj/W in T)
 		//Different classes of items give different commodities.
-		if (istype(W,/obj/item/weapon/cigbutt))
-			stored_comms["plastic"]++
-		else if(istype(W,/obj/effect/spider/spiderling))
-			stored_comms["wood"]++
-			stored_comms["wood"]++
-			stored_comms["plastic"]++
-			stored_comms["plastic"]++
-		else if(istype(W,/obj/item/weapon/light))
+		if (istype(W,/obj/item/organ))
+			playsound(src.loc, 'sound/effects/gib.ogg', 30, 1, -2)
+			src.loc.visible_message("\red [src.loc] sucks [W] into its decompiler. There's a horrible squelching noise.","\red You suck [W] into your decompiler. It makes a series of visceral crunching noises.")
```

Comment:
First off, color macros
>Text color macros

Use span classes, color macros are outdated.
`<span class='warning'>[text]</span>`

secondly, why is this message copy-pasted in three places? That's just bad.